ambry-cloud/src/main/java/com.github.ambry.cloud/azure/CosmosDataAccessor.java

```diff
@@ -155,6 +157,44 @@ void testConnectivity() {
     }
   }
 
+  /**
+   * Query Cosmos change feed to get the next set of {@code CloudBlobMetadata} objects in specified {@code partitionPath}
+   * after {@code requestContinationToken}, capped by specified {@code maxFeedSize} representing the max number of items to
+   * be queried from the change feed.
+   * @param requestContinationToken Continuation token after which change feed is requested.
+   * @param maxFeedSize max item count to be requested in the feed query.
+   * @param changeFeed {@link CloudBlobMetadata} {@code List} to be populated with the next set of entries returned by change feed query.
+   * @param partitionPath partition for which the change feed is requested.
+   * @return next continuation token.
+   * @throws DocumentClientException
+   */
+  public String queryChangeFeed(String requestContinationToken, int maxFeedSize, List<CloudBlobMetadata> changeFeed,
+      String partitionPath) throws DocumentClientException {
+    ChangeFeedOptions changeFeedOptions = new ChangeFeedOptions();
+    changeFeedOptions.setPartitionKey(new PartitionKey(partitionPath));
+    changeFeedOptions.setMaxItemCount(maxFeedSize);
+    if (requestContinationToken == null) {
+      changeFeedOptions.setStartFromBeginning(true);
+    } else {
+      changeFeedOptions.setRequestContinuation(requestContinationToken);
+    }
+    try {
+      FeedResponse<Document> feedResponse =
+          asyncDocumentClient.queryDocumentChangeFeed(cosmosCollectionLink, changeFeedOptions)
+              .limit(1)
+              .toBlocking()
+              .single();
```

Comment:
This will probably work for us, but if someone passed a high enough maxFeedSize to result in multiple feed pages, single() will throw an exception.  Maybe put a TODO/FIXME comment for now.